conda_forge_tick/update_sources.py

```diff
@@ -651,80 +653,125 @@ class NVIDIA(AbstractSource):
+            logger.debug("Found extra/feedstock-name.")
+            nvidia_compute_subdir = meta_yaml["meta_yaml"]["extra"]["feedstock-name"]
+        else:
+            logger.debug("Found the feedstock's name.")
+            nvidia_compute_subdir = meta_yaml["feedstock_name"]
+        logger.debug("The compute-subdir for NVIDIA URL is %s.", nvidia_compute_subdir)
+
+        logger.debug("Searching for a NVIDIA URL redist-json-name.")
+        if "redist-json-name" in meta_yaml["meta_yaml"]["extra"]:
+            logger.debug("Found explicit extra/compute-subdir.")
+            nvidia_redist_json_name = meta_yaml["meta_yaml"]["extra"][
+                "redist-json-name"
+            ]
+        elif "feedstock-name" in meta_yaml["meta_yaml"]["extra"]:
+            logger.debug("Found extra/feedstock-name.")
+            nvidia_redist_json_name = meta_yaml["meta_yaml"]["extra"]["feedstock-name"]
+        else:
+            logger.debug("Found the feedstock's name.")
+            nvidia_redist_json_name = meta_yaml["feedstock_name"]
+        logger.debug(
+            "The compute-subdir for NVIDIA URL is %s.", nvidia_redist_json_name
+        )
+
+        result = f"https://developer.download.nvidia.com/compute/{nvidia_compute_subdir}/redist#{nvidia_redist_json_name}"
+        logger.debug("The NVIDIA redist URL should be %s", result)
+        return result
 
     def get_version(self, url: str) -> Optional[str]:
-        return url  # = next version, same as in BaseRawURL
+        """Get latest version of a package by scraping nvidia.com JSONs.
+
+        url should be an address of the following form: "https://developer.download.
+        nvidia.com/compute/{nvidia_compute_subdir}/redist#{nvidia_redist_json_name}".
+        where we expect 'https://developer.download.nvidia.com/compute/
+        {{ nvidia_compute_subdir }}/redist/' to be a list of links which includes
+        multiple 'redistrib_X.Y.Z.json'. The latest of which will be the nominal latest
+        version. AND the string after # is the slug used to access the part of the JSON
+        blob which is relevant to the package.
+
+        Returns
+        -------
+            The full package version including build number.
+
+        Raises
+        ------
+            ValueError: If the slug is incorrect.
+        """
+        actual_url, slug = url.split("#")
+        logger.debug("Searching %s for redistrib_X.Y.Z.json", actual_url)
+        response = requests.get(actual_url)
+        html_content = response.text
+        # Search for links to redistrib_*.json in the response and strip the versions from there
+        # re doesn't support repeating patterns, so we assume there are three version numbers
+        redistrib_pattern = re.compile(
+            pattern=r"redistrib_[0-9]+\.[0-9]+\.[0-9]+\.json<"
+        )
+        result = re.findall(redistrib_pattern, html_content)
+        stripped_results = [x[10:-6] for x in result]
```

Comment:
Can we 
- strip filename extension `.json`
- split the string based on `_`
- get the version string from the second element 

?

Reply:
```suggestion
        stripped_results = [x.removesuffix(".json").split("_")[1] for x in result]
```